src/Compressors/Gaussian.jl

```diff
@@ -0,0 +1,159 @@
+    Each entry of ``S`` is generated independently following N(0, 1/s), namely, a 
+    Gaussian distribution with mean being zero and variance being 1 divided by the
+    compression dimension.
+    
+If ``A`` is compressed from the right, then we create a Gaussian sketch matrix, ``S``,
+    with dimension ``n \\times s``, where ``s`` is the compression dimension that 
+    is supplied by the user. 
+    Each entry of ``S`` is generated independently following N(0, 1/s), namely, a 
+    Gaussian distribution with mean being zero and variance being 1 divided by the 
+    compression dimension.
+
+# Fields
+
+- `cardinality::Type{<:Cardinality}`, the direction the compression matrix is intended to be 
+    applied to a target matrix or operator. Values allowed are `Left` or `Right`.
+- `compression_dim::Int64`, the target compression dimension. Referred to as ``s`` in the
+    mathematical description. 
+
+# Constructor
+
+    Gaussian(;cardinality=Left, compression_dim=2)
+
+## Arguments 
+- `cardinality::Type{<:Cardinality}`, the direction the compression matrix is intended to be 
+    applied to a target matrix or operator. Values allowed are `Left` or `Right`. By default 
+    `Left` is chosen. 
+- `compression_dim`, the target compression dimension. Referred to as ``s`` in the
+    mathemtical description. By default this is set to 2.
+
+## Returns 
+- A `Gaussian` object. 
+"""
+struct Gaussian <: Compressor
+    cardinality::Type{<:Cardinality}
+    compression_dim::Int64
+    # Check on the compression dimension
+    function Gaussian(cardinality, compression_dim)
+        if compression_dim < 0
+            throw(ArgumentError("Field 'Compression_dim' must be positive."))
+        end
+
+        return new(cardinality, compression_dim)
+    end
+end
+
+function Gaussian(; cardinality::Type{<:Cardinality}=Left, compression_dim::Int64=2)
+    # Partially construct the Gaussian datatype
+    return Gaussian(cardinality, compression_dim)
+end
+
+"""
+    GaussianRecipe <: CompressorRecipe
+
+The recipe containing all allocations and information for the Gaussian compressor.
+
+# Fields
+
+  - `cardinality::Type{<:Cardinality}`, the direction the compression matrix is intended to be
+    applied to a target matrix or operator. Values allowed are `Left` or `Right`.
+  - `n_rows::Int64`, the number of rows of the compression matrix.
+  - `n_cols::Int64`, the number of columns of the compression matrix.
+  - `scale::Number`, the standard deviation of Gaussian distribution during the compression matrix generation.
+  - `op::Matrix{Float64}`, the Gaussian compression matrix.
+"""
+mutable struct GaussianRecipe <: CompressorRecipe
+    cardinality::Type{<:Cardinality}
+    compression_dim::Int64
+    n_rows::Int64
+    n_cols::Int64
+    scale::Float64
+    op::Matrix{Float64}
+end
+
+function complete_compressor(
+    ingredients::Gaussian, A::AbstractMatrix, type::DataType=eltype(A)
```

Comment:
`type` here should not be an argument if `type` is specified as a field of `ingredients`